WordPress/gutenberg
# Fix and regenerate invalid block fixtures

Related: https://github.com/WordPress/gutenberg/pull/1929#issuecomment-318134153

This pull request seeks to resolve markup errors present in block fixtures to ensure that all fixtures are valid. As described in https://github.com/WordPress/gutenberg/pull/1929#issuecomment-318134153, we'll want some intentionally-invalid examples as well, but those are not included in the changes here.

The changes also add some nice conveniences to better compare a failed parse between its actual and expected representation. For example:

![Parse fail](https://user-images.githubusercontent.com/1779930/28682528-4e7e5dc2-72cb-11e7-82c2-a653300dd7c1.png)

This works only in Node environments currently, due to differences in log coloring between Node and browsers. This is something I'd like to consider improving, however separately.

There were a number of failed tests in `api/test/parser.js` after introducing these changes that I am surprised were ever working, since the dummy block implementations were clearly flawed. Those have been resolved here.

__Testing instructions:__

Verify that tests pass when regenerating fixtures from scratch:

```
./node_modules/.bin/rimraf blocks/test/fixtures/*.serialized.html && \
    ./node_modules/.bin/rimraf blocks/test/fixtures/*.json && \
    GENERATE_MISSING_FIXTURES=y npm run test-unit
```

Ensure there are no regressions with revised pullquote and gallery attribute defaults (moved to default attributes for consistency between `edit` and `save` usage).
